src/content_sao.cpp

```diff
@@ -1018,9 +1018,9 @@ void PlayerSAO::step(float dtime, bool send_recommended)
 		v3s16 p = floatToInt(getEyePosition(), BS);
 		MapNode n = m_env->getMap().getNodeNoEx(p);
 		const ContentFeatures &c = m_env->getGameDef()->ndef()->get(n);
-		// If player is alive & no drowning & not in ignore, breathe
-		if (m_breath < m_prop.breath_max &&
-				c.drowning == 0 && n.getContent() != CONTENT_IGNORE && m_hp > 0)
+		// If player is alive & not drowning & not in ignore & not immortal, breathe
+		if (m_breath < m_prop.breath_max && c.drowning == 0 &&
+				n.getContent() != CONTENT_IGNORE && m_hp > 0 && !isImmortal())
```

Comment:
`isImmortal()` check could be moved to the condition above in L1016.

Reply:
Good idea!